xla/service/collective_permute_decomposer.cc

```diff
@@ -142,13 +137,15 @@ absl::Status DecomposeCollectivePermute(
 
   HloInstruction* after_all =
       computation->AddInstruction(HloInstruction::CreateToken());
-  HloInstruction* recv = computation->AddInstruction(
-      HloInstruction::CreateRecv(data_shape, after_all, channel_id));
+  HloInstruction* recv = computation->AddInstruction(HloInstruction::CreateRecv(
+      /*shape=*/data_shape, /*token=*/after_all,
+      /*channel_id=*/channel_id.value_or(0)));
```

Comment:
Can you add a todo that we really want to make the channel ID optional in the op? 
Actively working on that

Reply:
Done